### ISSUE TITLE
[SRCCON Feedback] Create `partner` concept

### ISSUE THREAD

**Issue by mhkeller** [newslynx/opportunities#11]
Often newsrooms will have distribution partners with certain stories and they keep track of certain metrics about that partner and the stories' performance with them. This is related to the idea @abelsonlive and I were discussing about events having metrics as well. 

For instance, let's say your story is mentioned by someone, do you want to include that person's follower count as a metric? 

Or with this syndication example, if your story is picked up by another paper, can you bring in Meltwater's "reach" figures for that paper's homepage / do you want to start tracking total FB and Twitter shares on that URL? You'll need some way to associate those event metrics with a `partner` and an interface for ranking and slicing up partner metrics.

**Comment by mhkeller** [comment]
I believe we're currently tracking things like "A person's follower count from a tweet mentioning your work" in the event `meta` field but I don't believe that's super standardized.

**Comment by abelsonlive** [comment]
Yes, the `meta` field is super vague. It's basically a band-aid for the larger problem that it's hard to standardize data from different sources.

I think there are three approaches to the problem of event-level metrics:

##### 1. Create Sous Chefs which compute content metrics based off events.

An example of this would be: "compute the number of shares for articles that link to my article".  In this instance, you would have one sous chef that checks for trackbacks to your article from various sources and another that counts the number of shares for all of those urls.

The problem with this (the current) approach is that it precludes the ability to be able to do something like "show me the top events by share count that link to this content item" because we have no concept of event-level metrics. I suppose we could do something like create a "faceted" metric for a content item where the facets were event ids - this would be pretty hackish though.

##### 2. Create event-level metrics.

This actually wouldn't be too difficult. The framework for metrics is pretty general and could be applied to events without much hassle. From an interface perspective, what would that mean for you?

**Comment by abelsonlive** [comment]
sorry. i pressed enter too soon

##### 3. Partners

`Partners` seem like it could be encapsulated in an impact tag, no? Like, "this event signifies a partner republishing our article". I like this approach because it minifies the number of concepts in the API while encouraging use of tags - here, you might be able to calculate event-level metrics, but you'd need to filter the specific events you were interested in by their associated impact tags.

**Comment by mhkeller** [comment]
For **2**, the current metric concept is helpful in that it leaves open what those metrics are. I think different newsrooms will want to track different things and so leaving that up to them and "registering" a metric would be the way to go. I'm not quite sure how the interface would deal with this beyond more input fields, some of which could be prepopulated. Example use cases are (for partner pickups or localizations) "Did they include our video?" "Did they link back to us?"

Correct me if I'm wrong but could you write a metric that is triggered to run when an event is given a certain impact tag? That would say, "Ok, you just tagged this partner-pickup. I'm going to crawl that page and figure out whether it contains a link to us, whether it has a video that is ours (delegating that logic to the organization writing out these custom metrics).

I think short term, these will be filed out by hand. So I'm picturing a bunch of blank text fields, radio buttons or check boxes. So they would need a schema of some kind as well. It gets complicated.

For **3**, I think it might be more like, The Downtown Gazette is a "partner" and we want to see how many articles we've associated with them and what FB shares are on those articles vs articles we've published with The Uptown Gazette. Is this also what you were thinking. I could see that being an impact tag, like Media Promotion would make sense. If things get messy in the interface we could say that Media Promotion impact tags are akin to partners and they are displayed somewhat separately and a page could eventually be created to slice and dice articles by tags of media promotion.
